modules/siddhi-extensions/input-transports/jms-input-transport/src/main/java/org/wso2/siddhi/extension/input/transport/jms/util/JMSListener.java

```diff
@@ -0,0 +1,192 @@
+    private final JMSTaskManager jmsTaskManager;
+
+    public JMSListener(String listenerName, JMSTaskManager jmsTaskManager) {
+        this.listenerName = listenerName;
+        this.jmsTaskManager = jmsTaskManager;
+    }
+
+
+    /**
+     * Listen for JMS messages on behalf of the given service
+     */
+    private void start() {
+
+        JMSTaskManager stm = jmsTaskManager;
+        boolean connected = false;
+
+        /* The following parameters are used when trying to connect with the JMS provider at the start up*/
+        int r = 1;
+        long retryDuration = 10000;
+        double reconnectionProgressionFactor = 2.0;
+        long maxReconnectDuration = 1000 * 60 * 60; // 1 hour
+
+
+        // First we will check whether jms provider is started or not, as if not it will throw a continuous error log
+        // If jms provider not started we will wait for exponentially increasing time intervals,
+        // till the provider is started
+        while (!connected) {
+            boolean jmsProviderStarted = checkJMSConnection(stm);
+            if (jmsProviderStarted) {
+                log.info("Connection attempt: " + r + " for JMS Provider for listener: " + listenerName
+                        + " was successful!");
+                connected = true;
+                stm.start();
+
+                for (int i = 0; i < 3; i++) {
+                    // Check the consumer count rather than the active task count. Reason: if the
+                    // destination is of type topic, then the transport is only ready to receive
+                    // messages if at least one consumer exists. This is of not much importance,
+                    // except for automated tests.
+                    if (stm.getConsumerCount() > 0) {
+                        log.info("Started to listen on destination : " + stm.getDestinationJNDIName() +
+                                " of type " + JMSUtils.getDestinationTypeAsString(stm.getDestinationType()) +
+                                " for listener " + listenerName);
+                        return;
+                    }
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException ignore) {
+                    }
+                }
+
+                log.warn("Polling tasks on destination : " + stm.getDestinationJNDIName() +
+                        " of type " + JMSUtils.getDestinationTypeAsString(stm.getDestinationType()) +
+                        " for listener " + listenerName + " have not yet started after 3 seconds ..");
+            } else {
+                log.error("Unable to continue server startup as it seems the JMS Provider " +
+                        "is not yet started. Please start the JMS provider now.");
+                retryDuration = (long) (retryDuration * reconnectionProgressionFactor);
+                log.error("Connection attempt : " + (r++) + " for JMS Provider failed. Next retry in "
```

Comment:
Sorry meant to say use StringBuilder. Since this is an error log, it should be ok to use +.